### DISCUSSION TITLE
Scrape metrics from multiple resource groups

### DISCUSSION THREAD

**Discussion by wadhahbenaissa** [tomkerkhove/promitor#1598] [Q&A] [answered]
hello and I hope you are doing well!

I was wondering how can I scrape metrics from different resource groups?
- Is it that I have to create promitor instance per resource group?
- Is it that I have to define every time the same metric but with a different resource group (in the metrics declaration)
- Or is there an other alternative

Thanks

**Comment by tomkerkhove** [comment, marked as answer]
You can define a default resource group in the metric declaration ([docs](https://promitor.io/configuration/v2.x/metrics/#azure)), but can specify it for every resource that is being scraped as well.

In [this example](https://promitor.io/configuration/v2.x/metrics/#example) it will scrape `promitor-messaging` namespace in the `promitor` resource group, while `promitor-messaging-dev` is being scraped from `promitor-dev` resource group by specifying `resourceGroupName`

**Reply by wadhahbenaissa** [reply]
hello @tomkerkhove thank you so much for the quick interaction.
Actually, in my case, let's say I have 2 resource groups each containing several VMs, SAs .... and I am looking to get the relevant metrics for all the VMs and SAs from both resource groups.
And If I specify resourceGroupName in the metrics decalaration, then it will overwrite the resource group in the azure metadata section.
So, what would be the solution for this use case?

**Reply by tomkerkhove** [reply]
If you have multiple resources across RGs I'd recommend looking at setting up resource discovery ([concept](https://promitor.io/concepts/how-it-works#using-resource-discovery)) | [walkthrough](https://blog.tomkerkhove.be/2021/01/06/announcing-promitor-2-0-bringing-azure-monitor-metrics-where-you-need-them/#automatically-scraping-metrics-with-resource-discovery-in-promitor))

**Reply by wadhahbenaissa** [reply]
thanks for providing me with the links.
In fact, I am already using resource discovery, however I thought the resourceGroupName in azure metadata definition is required, which is not the case, right? Because I commented that line and redeployed promitor (s), and I am able now to fetch my SA metrics from different resource groups.

**Reply by tomkerkhove** [reply]
I don't know from the top of my head if the resource group name in the azure metadata definition is required but if it works I'm happy to hear that 👍

Can we mark the question as answered?

**Reply by wadhahbenaissa** [reply]
yes please and thank you so much again for your responses :)

**Reply by tomkerkhove** [reply]
Happy to help and it's the least I can do!

If you are using it in production, feel free to let me know and you can become a listed end-user!

**Comment by ayouba88** [comment]
Hi @tomkerkhove 

is there any update regarding the above answer? 
according to this answer, i do understand that if i have multiple subscriptions ( each subscriptions have 1 RG), so the way to scrape them is by creating scraper instance per Resource group. ( i am using discovery by the way) 

best regards,
Ayoub

**Reply by tomkerkhove** [reply]
Actually no, you can scrape multiple resources groups with one scraper and define the `azureMetadata.subscriptionId` and/or `azureMetadata.resourceGroupName` on a per-metric level. See https://docs.promitor.io/configuration/v2.x/metrics/#metrics

Hope that clarifies things a bit?

**Reply by ayouba88** [reply]
Thanks, it does clarify partially. but what i need, if i want to confguire it on the scraper level and not the metric level. is there a way to add multiple resource group names as an array ( as we can do with subscriptions Ids) ? ... i see that we don't have this option. therefore the options are
1- to confguire on metric level
2- create scraper per resource group + subscription

is that correct ?

**Reply by tomkerkhove** [reply]
No, `azureMetadata` are default values and cannot provide multiple of them because then I don't know to use :)

Options are:
1. Configure on metric level
2. Deploy scraper per resource group & subscription
3. Use resource discovery (recommended)
